zebra-state/src/service/non_finalized_state/chain.rs

```diff
@@ -937,6 +946,47 @@ impl Chain {
         }
     }
 
+    /// Returns the Orchard issued asset state if one is present in
+    /// the chain for the provided asset base.
+    pub fn issued_asset(&self, asset_base: &AssetBase) -> Option<AssetState> {
+        self.issued_assets.get(asset_base).cloned()
+    }
+
+    /// Remove the History tree index at `height`.
```

Comment:
@dmidem I think it would be simpler to keep the issued assets maps on each `ContextuallyValidatedBlock` partial relative to the previous block instead of making them partial relative to the finalized tip, but if they need to be relative to the finalized tip, copying the `AssetBase`s for every block would multiply the non-finalized state's theoretical max memory use, so those should go in `Arc`s.

```suggestion
    /// When a block with the provided `transactions` and partial `issued_assets` state is removed 
    /// from the chain tip, revert its changes from the [`Chain`]'s partial issued assets state.
    ///
    /// When a block is removed from the chain root to be committed to the finalized state, any
    /// assets in its partial `issued_assets` with an asset state matching the one in the chain's
    /// partial issued assets map is removed from the chain's partial issued assets map (since that
    /// it should match the finalized issued asset state.)
```